Properly parse energy states and occupations

pmrv opened the pull request:
Due to a faulty regex previously we threw away the signs of both energy states and occupations parsed from sphinx logs.

pmrv committed:
Properly parse energy states and occupations

Due to a faulty regex previously we threw away the signs of both energy
states and occupations parsed from sphinx logs.

niklassiemer commented on pyiron_atomistics/sphinx/base.py:
```diff
@@ -2124,7 +2124,7 @@ def n_steps(self):
         fa = re.findall(term, self.log_main, re.MULTILINE)
         arr = (
-            np.array(re.sub("[^0-9\. ]", "", "".join(fa)).split())
+            np.array(re.sub("[^-0-9\. ]", "", "".join(fa)).split())
```
Now this only works on negative quantities?!

niklassiemer replied:
wouldn't this allow, but not require a `-` sign? (Not tested!)
```suggestion
            np.array(re.sub("[^-?0-9\. ]", "", "".join(fa)).split())
```

pmrv replied:
No, this regex matches everything `[]` that is not `^` a `-`, `0-9`, `\.` (plain dot, `\` is for escaping) or ` ` and replaces it by the empty string.  This leaves just numbers and spaces, which are then "parsed".  Because the regex previously also matched the `-` the parsed number were effectively only the absolute value.

Actually the whole regex business is too complicated, because in the sphinx log the lines are of the form `some text: number number number`.  So removing that prefix would be enough.

niklassiemer replied:
Well in that case, it souds like an easier and more stable solution to replace everything until a colon (being greedy it will be the last colon)  by the empty string and parsing the numbers. Right now, a single `-` in the string would again flip the sign of the first number...

pmrv committed:
Use re capture groups

The weird double application of regex matching was only to remove a
(constant) prefix.  This is more elegantly done with capture groups.

pyiron-runner committed:
Format black
